### PR TITLE
Split range_check feature flag

### LINKED ISSUE #12543: Add the finalization logic flag for RangeCheck1 in Pickles

### PR DESCRIPTION
**Pull request by querolita**
Mina counterpart of proof-system's [967](https://github.com/o1-labs/proof-systems/pull/967)

It also modifies the lookups code by removing `foreign_field_add` and including `xor`. Please take a look at this in case this is not the right way to fix it (if at all).

* Closes #12543

### CODE REVIEW

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

File: src/lib/pickles/plonk_checks/plonk_checks.ml [hunk 1]
```diff
@@ -106,8 +106,8 @@ open Composition_types.Wrap.Proof_state.Deferred_values.Plonk
 
 type 'bool all_feature_flags =
   { lookup_tables : 'bool Lazy.t
-  ; table_width_1 : 'bool Lazy.t
-  ; table_width_2 : 'bool Lazy.t
+  ; table_width_at_least_1 : 'bool Lazy.t
```
Comment (mrmr1993) [inline src/lib/pickles/plonk_checks/plonk_checks.ml:109]: Nice :)

File: src/lib/pickles/wrap.ml [hunk 2]
```diff
@@ -566,7 +566,8 @@ let%test_module "gate finalization" =
 
         let actual_feature_flags =
           { Plonk_types.Features.none_bool with
-            range_check = true
+            range_check0 = true
```
Comment (mrmr1993) [inline src/lib/pickles/wrap.ml:569]: Should we also add a test that uses only `range_check0`?

**Review by mrmr1993 — APPROVED** [review]

**Comment by mrmr1993** [comment]
!ci-build-me

File: src/lib/pickles/wrap.ml (hunk 2 above)
Comment (querolita) [reply src/lib/pickles/wrap.ml:569]: Perhaps a 64-bit check alone? Sure, I will add it

**Comment by querolita** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by querolita** [comment]
!ci-build-me

**Comment by querolita** [comment]
!ci-build-me

**Comment by querolita** [comment]
!ci-build-me

File: src/app/test_executive/zkapps.ml [hunk 3]
```diff
@@ -36,7 +36,6 @@ module Make (Inputs : Intf.Test.Inputs_intf) = struct
           work_delay = Some 1
         ; transaction_capacity =
             Some Runtime_config.Proof_keys.Transaction_capacity.small
-        ; block_window_duration_ms = Some 150000
```
Comment (mimoo) [inline src/app/test_executive/zkapps.ml:39]: btw were the changes in this file supposed to happen? They seem unrelated

Comment (mimoo) [reply src/app/test_executive/zkapps.ml:39]: same question with a number of other files here : o

Comment (querolita) [reply src/app/test_executive/zkapps.ml:39]: Thanks for taking a look!

CI was giving an error that this branch couldn't merge cleanly into develop because it didn't know how to update zkapps.ml file (no idea why, because we did not make any changes on that file). Thus I brought the current version of this file in the develop branch so that it will be the same when the bump-for-lookup-features branch merges. But I can revert that change in the parent branch, or keep an eye open that the changes do not get affected when the final merge is produced.
